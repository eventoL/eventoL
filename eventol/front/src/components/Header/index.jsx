import React from 'react';
import PropTypes from 'prop-types';

import Logo from '../Logo';
import Navigation from '../Navigation';
import UserProfile from '../UserProfile';
import SignIn from '../SignIn';

import './index.scss';
import {LOGO_HEADER_DEFAULT} from '../../utils/constants';

export default class Header extends React.PureComponent {
  static propTypes = {
    handlerOnChangeLanguage: PropTypes.func.isRequired,
    isMobile: PropTypes.bool.isRequired,
    languages: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.string,
        name: PropTypes.string,
      })
    ),
    logoHeader: PropTypes.string,
    user: PropTypes.shape({
      first_name: PropTypes.string,
      last_name: PropTypes.string,
    }),
  };

  static defaultProps = {
    languages: [],
    logoHeader: LOGO_HEADER_DEFAULT,
    user: null,
  };

  handleToggle = event => {
    event.preventDefault();
    event.stopPropagation();
    const linksEl = document.querySelector('.narrow-links');
    const {
      style: {display},
    } = linksEl;
    linksEl.style.display = display === 'block' ? 'none' : 'block';
  };

  showUserIndicator = () => {
    const {user} = this.props;
    if (user) return <UserProfile user={user} />;
    return <SignIn />;
  };

  wideRender = () => {
    const {logoHeader, languages, handlerOnChangeLanguage} = this.props;
    return (
      <header className="header">
        <div className="nav-wide">
          <Logo logoHeader={logoHeader} />
          <Navigation
            handlerOnChangeLanguage={handlerOnChangeLanguage}
            languages={languages}
          />
          {this.showUserIndicator()}
        </div>
      </header>
    );
  };

  render() {
    const {
      isMobile,
      logoHeader,
      languages,
      handlerOnChangeLanguage,
    } = this.props;
    if (!isMobile) return this.wideRender();
    return (
      <header className="header">
        <div className="nav-narrow">
          <Logo logoHeader={logoHeader} />
          <div className="nav-right">
            <button
              onClick={this.handleToggle}
              onKeyPress={this.handleToggle}
              type="button"
            >
              <i className="fa fa-bars fa-2x" />
            </button>
            <div className="narrow-links">
              <Navigation
                handlerOnChangeLanguage={handlerOnChangeLanguage}
                languages={languages}
              />
              {this.showUserIndicator()}
            </div>
          </div>
        </div>
      </header>
    );
  }
}
