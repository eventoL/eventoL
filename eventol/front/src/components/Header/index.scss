@import '../../styles/index';

.header {
  @extend %flex-element;
  // FF3.6-15
  background: -webkit-linear-gradient(top, $gradient-color 0%, transparent 100%);

  // Chrome10-25, Safari5.1-6
  background: linear-gradient(to bottom, $gradient-color 0%, transparent 100%);

  box-sizing: border-box;

  // W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#a6000000', endColorstr='#00000000', GradientType=0);

  left: 0;
  padding: 40px;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 6;
}

.nav-wide {
  display: none;
}

@media (min-width: 950px) {
  .nav-wide {
    @include flex-element;
    box-sizing: border-box;
    width: 100vw;
  }

  .nav-narrow {
    display: none;
  }
}

@media (max-width: 950px) {
  .logo-component {
    position: absolute;
  }
}

.nav-narrow {
  @extend %flex-element;
  box-sizing: border-box;
  left: 0;
  padding: 40px;
  position: fixed;
  top: 0;
  width: 100vw;

  .nav-right {
    margin-left: auto;
    z-index: 7;

    button {
      background-color: transparent;
      border: 0;
      color: $white;
      display: contents;
      font-size: 100%;
      margin: 0;
      padding: 0;
      vertical-align: baseline;
    }

    i {
      float: right;
      z-index: 5;
    }

    .narrow-links {
      display: none;

      a {
        clear: right;
        display: inline;
      }
    }
  }
}

a {
  color: $a-color;
  text-decoration: none;

  &:focus,
  &:hover {
    color: $a-color;
    text-decoration: none;
  }
}
