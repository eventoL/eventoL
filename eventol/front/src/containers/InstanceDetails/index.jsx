import React from 'react';
import PropTypes from 'prop-types';
import withSizes from 'react-sizes';

import Hero from '../../components/Hero';
import Header from '../../components/Header';
import {mapSizesToProps} from '../../utils/dom';
import {
  BACKGROUND_DEFAULT,
  LOGO_HEADER_DEFAULT,
  LOGO_LANDING_DEFAULT,
} from '../../utils/constants';

import './index.scss';

class InstanceDetails extends React.PureComponent {
  static propTypes = {
    background: PropTypes.string,
    events: PropTypes.number.isRequired,
    handleOnChangeLanguage: PropTypes.func,
    isMobile: PropTypes.bool.isRequired,
    languages: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.string,
        name: PropTypes.string,
      })
    ),
    logoHeader: PropTypes.string,
    logoLanding: PropTypes.string,
    user: PropTypes.shape({
      first_name: PropTypes.string,
      last_name: PropTypes.string,
    }),
    users: PropTypes.number.isRequired,
    versions: PropTypes.shape({
      commit: PropTypes.string.isRequired,
      django: PropTypes.string.isRequired,
      python: PropTypes.string.isRequired,
      tag: PropTypes.string.isRequired,
    }).isRequired,
  };

  static defaultProps = {
    background: BACKGROUND_DEFAULT,
    handleOnChangeLanguage: null,
    languages: [],
    logoHeader: LOGO_HEADER_DEFAULT,
    logoLanding: LOGO_LANDING_DEFAULT,
    user: null,
  };

  handlerOnChangeLanguage = languageCode => {
    const {handleOnChangeLanguage} = this.props;
    if (handleOnChangeLanguage) {
      handleOnChangeLanguage(languageCode);
    }
  };

  render() {
    const {
      background,
      isMobile,
      languages,
      logoHeader,
      logoLanding,
      user,
      users,
      events,
      versions,
    } = this.props;
    return (
      <div>
        <Header
          handlerOnChangeLanguage={this.handlerOnChangeLanguage}
          isMobile={isMobile}
          languages={languages}
          logoHeader={logoHeader}
          user={user}
        />
        <Hero
          background={background}
          isLogged={false}
          logoLanding={logoLanding}
          message={`EventoL ${versions.tag}`}
        >
          <h6>
            {`Users: ${users}`}
            <br />
            {`Events: ${events}`}
            <br />
            {`Django: ${versions.django}`}
            <br />
            {`Python: ${versions.python}`}
            <br />
            {`Commit: ${versions.commit}`}
          </h6>
        </Hero>
      </div>
    );
  }
}

export default withSizes(mapSizesToProps)(InstanceDetails);
