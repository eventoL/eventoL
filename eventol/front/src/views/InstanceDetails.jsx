import React from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';

import getStore from '../store';
import InstanceDetails from '../containers/InstanceDetails';

import './index.scss';

const store = getStore();

window.render_components = properties => {
  window.params = {...properties};
  render(
    <Provider store={store}>
      <InstanceDetails
        background={properties.background}
        eventolMessage={properties.message}
        events={properties.events}
        handleOnChangeLanguage={properties.onChangeLanguage}
        languages={properties.languages}
        logoHeader={properties.logo_header}
        logoLanding={properties.logo_landing}
        user={properties.user}
        users={properties.users}
        versions={properties.versions}
      />
    </Provider>,
    document.getElementById('root')
  );
};

if (module.hot) {
  if (window.params) window.render_components(window.params);
  module.hot.accept();
}
