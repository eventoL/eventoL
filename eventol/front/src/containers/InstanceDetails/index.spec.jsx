import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../components/Hero', () => 'Hero');
jest.mock('../../components/Header', () => 'Header');
jest.mock('../../components/Search', () => 'Search');
jest.mock('../../components/TitleList', () => 'TitleList');

import Header from '../../components/Header';

import InstanceDetails from '.';

describe('InstanceDetails', () => {
  let background;
  let component;
  let element;
  let eventolMessage;
  let events;
  let instance;
  let isMobile;
  let languages;
  let logoHeader;
  let logoLanding;
  let props;
  let tree;
  let user;
  let users;
  let versions;
  const handleOnChangeLanguage = jest.fn();

  const getComponent = allProps => {
    element = <InstanceDetails {...allProps} />;
    component = renderer.create(element);
    instance = component.root;
    return component;
  };

  beforeEach(() => {
    isMobile = false;
    background = 'background';
    logoHeader = 'logoHeader';
    logoLanding = 'logoLanding';
    events = 500;
    eventolMessage = 'eventolMessage';
    user = {
      first_name: 'first_name',
      last_name: 'last_name',
    };
    languages = [
      {code: 'es', name: 'Spanish'},
      {code: 'en', name: 'English'},
    ];
    users = 4000;
    versions = {
      commit: '3af85e99036166d1267c3794e333a5b12c729561 Fix react lints',
      django: '1.11.29',
      python: '3.9.6 (default, Jun 29 2021, 19:45:46) [GCC 10.2.1 20201203]',
      tag: 'v2.3.0',
    };

    props = {
      background,
      eventolMessage,
      events,
      handleOnChangeLanguage,
      isMobile,
      logoHeader,
      logoLanding,
      user,
      users,
      versions,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('handlerOnChangeLanguage', () => {
    beforeEach(() => {
      component = getComponent({
        ...props,
        isMobile: false,
        languages,
        handleOnChangeLanguage,
      });
      tree = component.toJSON();
    });

    test('should calls handleOnChangeLanguage with language code when this function exists', async () => {
      const HeaderInstance = instance.findByType(Header);
      HeaderInstance.props.handlerOnChangeLanguage('es');

      expect(handleOnChangeLanguage).toBeCalled();
      expect(handleOnChangeLanguage).toBeCalledWith('es');
    });

    test('should not calls handleOnChangeLanguage when this function not exists', async () => {
      component = getComponent({
        ...props,
        handleOnChangeLanguage: null,
        isMobile: false,
        languages,
      });

      const HeaderInstance = instance.findByType(Header);
      HeaderInstance.props.handlerOnChangeLanguage('es');

      expect(handleOnChangeLanguage).not.toBeCalled();
    });
  });

  describe('With languages', () => {
    test('Snapshot', () => {
      component = getComponent({
        ...props,
        isMobile: false,
        languages,
      });
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('With user', () => {
    describe('Desktop', () => {
      beforeEach(() => {
        component = getComponent({...props, user, isMobile: false});
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('Mobile', () => {
      beforeEach(() => {
        component = getComponent({...props, user, isMobile: true});
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('Without user', () => {
    describe('Desktop', () => {
      beforeEach(() => {
        component = getComponent({...props, isMobile: false});
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });

    describe('Mobile', () => {
      beforeEach(() => {
        component = getComponent({...props, isMobile: true});
        tree = component.toJSON();
      });

      test('Snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
