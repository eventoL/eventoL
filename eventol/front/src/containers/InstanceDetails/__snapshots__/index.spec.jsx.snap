// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceDetails With languages Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={
      Array [
        Object {
          "code": "es",
          "name": "Spanish",
        },
        Object {
          "code": "en",
          "name": "English",
        },
      ]
    }
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="EventoL v2.3.0"
  >
    <h6>
      Users: 4000
      <br />
      Events: 500
      <br />
      Django: 1.11.29
      <br />
      Python: 3.9.6 (default, Jun 29 2021, 19:45:46) [GCC 10.2.1 20201203]
      <br />
      Commit: 3af85e99036166d1267c3794e333a5b12c729561 Fix react lints
    </h6>
  </Hero>
</div>
`;

exports[`InstanceDetails With user Desktop Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="EventoL v2.3.0"
  >
    <h6>
      Users: 4000
      <br />
      Events: 500
      <br />
      Django: 1.11.29
      <br />
      Python: 3.9.6 (default, Jun 29 2021, 19:45:46) [GCC 10.2.1 20201203]
      <br />
      Commit: 3af85e99036166d1267c3794e333a5b12c729561 Fix react lints
    </h6>
  </Hero>
</div>
`;

exports[`InstanceDetails With user Mobile Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={true}
    languages={Array []}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="EventoL v2.3.0"
  >
    <h6>
      Users: 4000
      <br />
      Events: 500
      <br />
      Django: 1.11.29
      <br />
      Python: 3.9.6 (default, Jun 29 2021, 19:45:46) [GCC 10.2.1 20201203]
      <br />
      Commit: 3af85e99036166d1267c3794e333a5b12c729561 Fix react lints
    </h6>
  </Hero>
</div>
`;

exports[`InstanceDetails Without user Desktop Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={false}
    languages={Array []}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="EventoL v2.3.0"
  >
    <h6>
      Users: 4000
      <br />
      Events: 500
      <br />
      Django: 1.11.29
      <br />
      Python: 3.9.6 (default, Jun 29 2021, 19:45:46) [GCC 10.2.1 20201203]
      <br />
      Commit: 3af85e99036166d1267c3794e333a5b12c729561 Fix react lints
    </h6>
  </Hero>
</div>
`;

exports[`InstanceDetails Without user Mobile Snapshot 1`] = `
<div>
  <Header
    handlerOnChangeLanguage={[Function]}
    isMobile={true}
    languages={Array []}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    isLogged={false}
    logoLanding="logoLanding"
    message="EventoL v2.3.0"
  >
    <h6>
      Users: 4000
      <br />
      Events: 500
      <br />
      Django: 1.11.29
      <br />
      Python: 3.9.6 (default, Jun 29 2021, 19:45:46) [GCC 10.2.1 20201203]
      <br />
      Commit: 3af85e99036166d1267c3794e333a5b12c729561 Fix react lints
    </h6>
  </Hero>
</div>
`;
